### DISCUSSION TITLE
Is it possible to pass the MongoDB version for download without modifying the package.json?

### DISCUSSION THREAD

**Discussion by nodermann2** [typegoose/mongodb-memory-server#590] [Q&A] [answered]
## Versions

- NodeJS: 16.13.1
- mongodb-memory-server-*: 6.9.3
- mongodb: 4.0.16
- system: Ubuntu 20.04.3 LTS

package: mongo-memory-server

## What is your question?
Is it possible to pass the MongoDB version for download without modifying the package.json? as stated here #316
Our CircleCI tests access a non-existent path https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-debian10-4.0.16.tgz and cannot download the .tgz

We use https://github.com/mikhail-angelov/mongo-unit which uses your mongodb-memory-server and we would not like to use forks because they may contain build errors. The solution of transferring the desired link/version to MongoDB via env would be the best of all.

**Comment by hasezoey** [comment, marked as answer]
> Is it possible to pass the MongoDB version for download without modifying the package.json?

yes with environment variables, just prefix any config option at [Config Options](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options/) with `MONGOMS_` as [explained here](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options/#how-to-use-them-as-environment-variables)

----

in your case you may want to overwrite the full download url, so use Config Options [`DOWNLOAD_URL`](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options/#download_url) like `MONGOMS_DOWNLOAD_URL=https://your.com/path/to/the/archive.tgz`

---

PS: you may want to upgrade the MMS version if possible, it is known that `4.0.16` is not existing for all platforms anymore, and later versions use higher versions already, as can be [read here](https://nodkz.github.io/mongodb-memory-server/docs/guides/mongodb-server-versions#mongodb-memory-server-core-version-table)
